### PR TITLE
Solve identical comparison with local-variance

### PR DESCRIPTION
**Pull request by VincentLanglet**
Close https://github.com/phpstan/phpstan/issues/11161
Close https://github.com/phpstan/phpstan/issues/10697

### CODE REVIEW

**Comment by staabm** [comment]
Issue-bot reported this seem to fix also another issue: https://github.com/phpstan/phpstan-src/actions/runs/9484392325

Please add a regression test

File: src/Reflection/InitializerExprTypeResolver.php [hunk 1]
```diff
@@ -1345,8 +1349,9 @@ public function resolveIdenticalType(Type $leftType, Type $rightType): BooleanTy
 			return new ConstantBooleanType($leftTypeFiniteTypes[0]->equals($rightTypeFiniteType[0]));
 		}
 
-		$isSuperset = $leftType->isSuperTypeOf($rightType);
-		if ($isSuperset->no()) {
+		$isLeftSuperset = $leftType->isSuperTypeOf($rightType);
```
Comment (ondrejmirtes) [inline src/Reflection/InitializerExprTypeResolver.php]: SuperType, not Superset

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Yeah, that regression test too please.

**Comment by VincentLanglet** [comment]
> Yeah, that regression test too please.

Done

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**